keps/sig-node/2411-cri-container-log-rotation/README.md

```diff
@@ -0,0 +1,301 @@
+
+_This section must be completed when targeting alpha to a release._
+
+- **How can this feature be enabled / disabled in a live cluster?**
+  - [x] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name: CRIContainerLogRotation
+    - Components depending on the feature gate: Kubelet
+  - [ ] Other
+    - Describe the mechanism:
+    - Will enabling / disabling the feature require downtime of the control
+      plane?
+    - Will enabling / disabling the feature require downtime or reprovisioning
+      of a node? (Do not assume `Dynamic Kubelet Config` feature is enabled).
+
+- **Does enabling the feature change any default behavior?**
+
+  With the dockershim, the docker daemon was in charge of managing and rotating the logs. With a CRI container runtime, the kubelet is in charge of managing and rotating logs. There is no real change to default behavior apart form the fact that the log rotation will depend on which container runtime is being used.
+
+- **Can the feature be disabled once it has been enabled (i.e. can we roll back
+  the enablement)?**
+
+  Yes, but if disabled the container logs will not be rotated when using a CRI container runtime.
+
+- **What happens if we reenable the feature if it was previously rolled back?**
+
+  No impact, container log rotation will work for CRI container runtimes.
+
+- **Are there any tests for feature enablement/disablement?**
+
+  There are already unit and node e2e tests in place for this feature.
+
+### Rollout, Upgrade and Rollback Planning
+
+_This section must be completed when targeting beta graduation to a release._
+
+- **How can a rollout fail? Can it impact already running workloads?**
+
+   When the container log manager doesn't start up and rotate the logs as expected. Restarts shouldn't affect ths as it is the container runtime that will be writing the logs. The kubelet is in charge of checking log size and rotating when needed.
+
+- **What specific metrics should inform a rollback?**
+
+  - There is major loss of logs on nodes that use a CRI runtime.
+  - Logging agents are unable to integrate with k8s and aggregate logs to a remote storage.
+
+- **Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?**
+  
+  Any manual testing was done when the feature was initially implemented in https://github.com/kubernetes/kubernetes/pull/59898.
+
+- **Is the rollout accompanied by any deprecations and/or removals of features, APIs, 
+fields of API types, flags, etc.?**
+  
+  No
+
+### Monitoring Requirements
+
+_This section must be completed when targeting beta graduation to a release._
+
+- **How can an operator determine if the feature is in use by workloads?**
+
+  When a CRI container runtime is used, the logs are being rotated and stored in the log directory structure with a gzip format.
+
+- **What are the SLIs (Service Level Indicators) an operator can use to determine 
+the health of the service?**
+  - [ ] Metrics
```

Comment:
should there be a metric on logs file size?

Reply:
i would argue against a metric specific to logs.  log usage is really charged to the per container ephemeral storage usage, and we already can evict/cleanup pods that exhaust ephemeral storage as part of eviction manager.

Reply:
> i would argue against a metric specific to logs. log usage is really charged to the per container ephemeral storage usage, and we already can evict/cleanup pods that exhaust ephemeral storage as part of eviction manager.

This would be useful information to have in the KEP.  It sounds like that would be a thing to watch generally.